projects/eventonica/eventonica-part7-react.md

```diff
@@ -72,10 +70,145 @@ Here's an example of what `package.json` might look like now:
+
+Before this stage, ensure you have a commit in place with the working app.
+
+Here is example code that might be similar to the way you displayed a list of users in your app. Depending on how you built Part 2, your code might not be in a function like this but hopefully you can translate the concept to your own code.
+
+```js
+// ----- USERS ----- //
+// Display all users when app is loaded
+function displayUsers() {
+  let displayUserText = '';
+  for (let user of eventonica.allUsers) {
+    displayUserText += `<li>${user.userName}, ID: ${user.userID}</li>`;
+  }
+  document.querySelector('#all-users').innerHTML = displayUserText;
+}
+```
+
+If you think about how React works, this can be represented by a component like this:
+
+```jsx
+// src/UsersList.js
+class UsersList extends React.Component {
+  render() {
+    return (
+      <div id="all-users">
+        {eventonica.allUsers.map((user) => (
+          <li>
+            {user.userName}, ID: {user.userID}
+          </li>
+        ))}
+      </div>
+    );
+  }
+}
+```
+
+Try migrating your version of the users list and see if you can get it to display. Note you're still not using props yet, but we're trying to move in small increments to make our progress more manageable.
+
+### Use React props to pass data
+
+Instead of globally referencing `eventonica`, we want to make our app follow the React pattern of using props.
+
+```jsx
+// src/index.js - where <App> is created and mounted to the DOM
+import Eventonica from './models.js';
+
+const eventonica = new Eventonica();
+eventonica.addUser('Lisa', 12345);
+// ... any other starter data you want
+
+ReactDOM.render(
+  <React.StrictMode>
+    <App eventonica={eventonica} />
+  </React.StrictMode>,
+  document.getElementById('root')
+);
+
+// src/App.js
+class App extends React.Component {
+  render() {
+    const { eventonica } = this.props;
+    return (
+      <div>
+        <UsersList users={eventonica.allUsers} />
+      </div>
+    );
+  }
+}
+```
+
+Now you will have to modify `UsersList.js` to accept this prop. If you change the seed data above, CRA should live reload with the changes.
+
+### Event Handling
+
+In your Part 2 version, you attached event handlers to the DOM. With React, you will use props like `onClick` to achieve the same result. Again, this example code may not be exactly how your code works but hopefully you can see how to apply the idea to yours.
+
+#### JS DOM Example of Handling a User Delete Action
+
+```js
+document
+  .querySelector('#delete-user-action')
+  .addEventListener('click', (event) => {
+    event.preventDefault();
+    const idToDelete = parseInt(
+      document.querySelector('#delete-user-id').value
+    );
+    eventonica.deleteEvent(idToDelete);
+    displayEvents(); // refresh the DOM
+  });
+```
+
+In React we would setup the event handler like this:
+
+```jsx
+// src/UserDeleteForm.js
+class UserDeleteForm extends React.Component {
+  state = { idToDelete: null };
```

Comment:
Thanks for doing this!  Again, I'm not sure this is the way to initiate state... I think you need a constructor and everything.